### DISCUSSION TITLE
Bug or user error?

### DISCUSSION THREAD

**Discussion by Algorab** [Ogeon/palette#253] [Q&A] [answered]
Try to convert from the hsl colorspace to the rgb colorspace and don't get the
expected values. Not sure is it a bug or a user error.

cargo play script
``` 
//# palette = "0.6.0"
use palette;
use palette::{Hsl, RgbHue, FromColor};
use palette::rgb::Rgb;

fn main() {

    let degrees = 60.0;
    let saturation = 100.0;
    let lightness = 50.0;

    let hsl: Hsl = Hsl::from_components((RgbHue::from_degrees(degrees), saturation, lightness));
    let rgb: Rgb = Rgb::from_color(hsl);

    println!("{:?}", rgb);

    //Expect RGB values for yellow
    assert_eq!(rgb.red, 255.0, "expect for red 255");
    assert_eq!(rgb.green, 255.0, "expect for green 255");
    assert_eq!(rgb.blue, 0.0, "expect for blue 0");

}

```

**Comment by Ogeon** [comment, marked as answer]
It looks to me like it's just a matter of using different scales (there are multiple conventions). The ranges of RGB, saturation and lightness is 0.0 to 1.0 when using `f32` and `f64`. So if I'm reading this correctly, saturation should be 1.0, lightness should be 0.5 and the expected RGB values either changed from 255.0 to 1.0 or converted to `u8` with some variant of `rgb.into_format::<u8>()`.

You can check the documentation for the ranges of each color space.

**Reply by Algorab** [reply]
You are completely right. After applying what you mentioned, it works as I expect. 
Thanks a lot.
